berkdb/env/env_open.c

```diff
@@ -927,6 +931,10 @@ __dbenv_refresh(dbenv, orig_flags, rep_check)
 	 * log file handles.  Ick.
 	 */
 	LIST_INIT(&dbenv->dblist);
+    if (dbenv->fileidhash != NULL) {
+        hash_clear(dbenv->fileidhash);
```

Comment:
should we also call hash_free() ?

Reply:
hash_clear() seems to do almost same work as hash_free() so it should be enough.

Reply:
There are not the same; hash_clear() empty a hash (frees elements stored inside), while hash_free() does a hash_clear() followed by freeing the hash itself.